desc/objectives/getters.py

```diff
@@ -214,20 +216,30 @@ def get_NAE_constraints(
     return constraints
 
 
-def maybe_add_self_consistency(eq, constraints):
+def maybe_add_self_consistency(thing, constraints):
     """Add self consistency constraints if needed."""
 
     def _is_any_instance(things, cls):
         return any([isinstance(t, cls) for t in things])
 
-    if not _is_any_instance(constraints, BoundaryRSelfConsistency):
-        constraints += (BoundaryRSelfConsistency(eq=eq),)
-    if not _is_any_instance(constraints, BoundaryZSelfConsistency):
-        constraints += (BoundaryZSelfConsistency(eq=eq),)
-    if not _is_any_instance(constraints, FixLambdaGauge):
-        constraints += (FixLambdaGauge(eq=eq),)
-    if not _is_any_instance(constraints, AxisRSelfConsistency):
-        constraints += (AxisRSelfConsistency(eq=eq),)
-    if not _is_any_instance(constraints, AxisZSelfConsistency):
-        constraints += (AxisZSelfConsistency(eq=eq),)
+    # Equilibrium
+    if hasattr(thing, "_Rb_lmn") and hasattr(thing, "_Zb_lmn"):
```

Comment:
Is this a good enough way to check for class types? I didn't want to use `isinstance` to avoid circular imports.

Reply:
Maybe checking for `"_Psi"` would be better for Equilibrium? Although I think the surface classes use `"_R_lmn"` instead of `"Rb_lmn"`